src/all/comikey/src/eu/kanade/tachiyomi/extension/all/comikey/Comikey.kt

```diff
@@ -0,0 +1,389 @@
+                .body
+                .string(),
+        )
+
+        return data.episodes
+            .filter { !it.availability.purchaseEnabled || !hidePaidChapters }
+            .map {
+                SChapter.create().apply {
+                    url = "/read/$mangaSlug/${makeEpisodeSlug(it, defaultChapterPrefix)}/"
+                    name = buildString {
+                        append(it.title)
+
+                        if (it.subtitle != null) {
+                            append(": ")
+                            append(it.subtitle)
+                        }
+                    }
+                    chapter_number = it.number
+                    date_upload = try {
+                        dateFormat.parse(it.releasedAt)!!.time
+                    } catch (e: Exception) {
+                        0L
+                    }
+                }
+            }
+            .reversed()
+    }
+
+    override fun chapterListSelector() = throw UnsupportedOperationException()
+
+    override fun chapterFromElement(element: Element) = throw UnsupportedOperationException()
+
+    override fun fetchPageList(chapter: SChapter): Observable<List<Page>> {
+        return Observable.fromCallable {
+            pageListParse(chapter)
+        }
+    }
+
+    override fun pageListParse(document: Document) = throw UnsupportedOperationException()
+
+    @SuppressLint("SetJavaScriptEnabled")
+    private fun pageListParse(chapter: SChapter): List<Page> {
+        val interfaceName = randomString()
+
+        val handler = Handler(Looper.getMainLooper())
+        val latch = CountDownLatch(1)
+        val jsInterface = JsInterface(latch, json)
+        var webView: WebView? = null
+
+        handler.post {
+            val innerWv = WebView(Injekt.get<Application>())
+
+            webView = innerWv
+            innerWv.settings.domStorageEnabled = true
+            innerWv.settings.javaScriptEnabled = true
+            innerWv.settings.blockNetworkImage = true
+            innerWv.setLayerType(View.LAYER_TYPE_SOFTWARE, null)
+            innerWv.addJavascriptInterface(jsInterface, interfaceName)
+
+            innerWv.webViewClient = object : WebViewClient() {
+                override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
+                    super.onPageStarted(view, url, favicon)
+                    view?.evaluateJavascript(webviewScript.replace("__interface__", interfaceName)) {}
+                }
+            }
+
+            innerWv.loadUrl("$baseUrl${chapter.url}")
+        }
+
+        latch.await(30, TimeUnit.SECONDS)
+        handler.post { webView?.destroy() }
+
+        if (latch.count == 1L) {
+            throw Exception(intl["error_timed_out_decrypting_image_links"])
+        }
+
+        if (jsInterface.error.isNotEmpty()) {
+            val message = if (jsInterface.error.startsWith("[")) {
+                val key = jsInterface.error.substringAfter("[").substringBefore("]")
```

Comment:
You should either replace all instances of stuff enclosed in square brackets, or assume the whole string should be replaced if it starts with a square bracket, not both at once.

`if` condition implies the key starts at char 1
`key =` implies that there is one key somewhere in the string

Reply:
Replaced with a proper i18n system for the WebView script.